65479 update content needs help all pages

lee-delarm6 opened the pull request:
## Summary

- Update content section in the help block
- Added new help block to refactor some older code
- Removed some sections per UX design of help block
- Display help block on all appropriate pages
- Links all work

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/65479
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/66618

## Testing done

- Removed old content sections for help
- Added new help block named `<MainHelpBlock />`

## Screenshots

<img width="469" alt="Screenshot 2023-10-03 at 10 32 55 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/135059743/dbcdd027-c59f-4dbc-9604-0a7443fb795e">
<img width="995" alt="Screenshot 2023-10-03 at 10 32 45 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/135059743/9c12fa90-d411-4f72-ac48-4f7c3e1cf340">
<img width="505" alt="Screenshot 2023-10-03 at 10 33 19 AM" src="https://github.com/department-of-veterans-affairs/vets-website/assets/135059743/e969b0ad-502e-4091-974b-14d8e9a2766f">


## Quality Assurance & Testing

- [x] I updated and fixed all unit tests
- [x] I updated and fixed all cypress tests

lee-delarm6 committed:
Added main block and changed display

Added main block and changed when it displays

lee-delarm6 committed:
Content update for tests, no more extra info

Content update for tests, no more extra info

brianseek commented on src/applications/check-in/day-of/tests/e2e/pages/Confirmation.js:
```diff
@@ -97,12 +97,6 @@ class Confirmation {
       );
   };
 
-  validateExtraFooterMessage = () => {
```
Why was this test removed?

lee-delarm6 replied:
"day-of-travel-extra-message" no longer exists since the help block refactored some of this code into a single help block component, and it was asked that some of the extra content be removed in the process as well in the ticket

brianseek reviewed (changes requested):
The new component MainHelpBlock needs a unit test. I think there needs to be testing in the footer around the logic to show different help messages as well.

I know that the TravelHelpBlock existed before this PR, but I really think it and the MainHelpBlock have too much duplicated code/content. It would be better to have a single component that combines these two and reduces the duplicated code/content.

lee-delarm6 commented:
> The new component MainHelpBlock needs a unit test. I think there needs to be testing in the footer around the logic to show different help messages as well.
> 
> I know that the TravelHelpBlock existed before this PR, but I really think it and the MainHelpBlock have too much duplicated code/content. It would be better to have a single component that combines these two and reduces the duplicated code/content.

I think I prefer the split components because the logic to show or hide each one is held in the footer, and I'd rather not move that responsibility to the component itself (if it were singular). A lot of the testing remains intact since I moved the main portion of the code into the `<MainHelpBlock>` but kept the `pre-check-in-message` encapsulating it. It may need updated unit tests though as you said

I might also just not be looking at it in the right view, will take a fresh look tomorrow morning

lee-delarm6 committed:
Added unit tests for mainhelpblock

Added unit tests for mainhelpblock

lee-delarm6 commented:
@brianseek Added a unit test for the mainhelpblock

lee-delarm6 committed:
Merge branch 'main' into 65479_update_content_needs_help_all_pages

lee-delarm6 committed:
Merge branch 'main' into 65479_update_content_needs_help_all_pages

lee-delarm6 committed:
Refactor helpblock, add tests, fix tests

Refactored mainhelp and travelhelp block into single block, updated all tests, fixed breaking ones

brianseek commented on src/applications/check-in/components/layout/tests/Footer.unit.spec.jsx:
```diff
@@ -100,5 +85,17 @@ describe('check-in', () => {
       );
       expect(component.getByTestId('travel-help-block')).to.exist;
     });
+    it('Renders HelpBlock for complete page', () => {
```
This is describing the complete page but not testing the complete page. I think just the `it` descriptor needs to be updated to describe this test.

brianseek commented on src/applications/check-in/components/layout/tests/Footer.unit.spec.jsx:
```diff
@@ -67,7 +52,7 @@ describe('check-in', () => {
       );
       expect(component.getByTestId('travel-help-block')).to.exist;
```
this is testing a testId that is always present no matter if it is showing the travel info

brianseek commented on src/applications/check-in/components/layout/Footer.jsx:
```diff
@@ -29,49 +27,14 @@ const Footer = ({ router, isPreCheckIn }) => {
-            <va-telephone contact="711" tty />.
-          </p>
+      {travelPages.includes(currentPage) ||
+      (currentPage && currentPage.includes('complete')) ? (
```
this logic should exclude the complete page for pre-check-in.  You could bring back the isPreCheckIn prop or select the app from redux.

brianseek reviewed (changes requested):
looking good! some cleanup needed in the tests and one testId that needs to be conditional or removed and use a different identifier. Also pre-check-in complete page is showing the travel footer when it shouldn't

lee-delarm6 committed:
Update naming and check

Complete page only shows if check-in, updated naming

lee-delarm6 commented:
> looking good! some cleanup needed in the tests and one testId that needs to be conditional or removed and use a different identifier. Also pre-check-in complete page is showing the travel footer when it shouldn't

Thanks! I was mixed up on the pre-check-in vs check-in, added the isPrecheckin flag back and fixed that as well as the other comments mentioned! @brianseek

brianseek commented on src/applications/check-in/components/layout/tests/Footer.unit.spec.jsx:
```diff
@@ -87,9 +74,22 @@ describe('check-in', () => {
+      expect(component.getByTestId('for-questions-about-filing')).to.exist;
     });
-    it('Renders TravelHelpBlock for complete page', () => {
+    it('Renders HelpBlock for complete page pre-check and travel section', () => {
```
this is a little confusing to me. This is this a pre-check-in test or day-of? The Provider sets the app to pre-check-in but the footer says its not pre-check-in and then you are testing for day-of check-in things. I would refine this one a bit.

brianseek reviewed (approved):
One little thing to clear up confusion in a test then this is good to go. Dropping an approval, so you can merge after sorting that last thing out. 🥇

brianseek commented on src/applications/check-in/components/layout/tests/Footer.unit.spec.jsx:
```diff
@@ -98,7 +98,19 @@ describe('check-in', () => {
-      expect(component.getByTestId('travel-help-block')).to.exist;
+      expect(component.getByTestId('help-block')).to.exist;
+    });
+    it('Renders HelpBlock for contact-information page', () => {
```
another suggestion for describing this test could be "Renders HelpBlock without travel link on non-travel pages"

lee-delarm6 replied:
True, since that's what I'm really testing for here

lee-delarm6 committed:
More concise test wording

More concise test wording
